qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp

```diff
@@ -36,15 +38,17 @@ Mantid::Kernel::Logger g_log("Reflectometry GUI");
  * @param view :: [input] The view we are managing
  * @param messageHandler :: Interface to a class that displays messages to
  * the user
+ * @param slitCalculator :: Interface to the Slit Calculator dialog
  * @param batchPresenterFactory :: [input] A factory to create the batches
  * we will manage
  */
 MainWindowPresenter::MainWindowPresenter(
     IMainWindowView *view, IMessageHandler *messageHandler,
+    std::unique_ptr<ISlitCalculator> slitCalculator,
     std::unique_ptr<IBatchPresenterFactory> batchPresenterFactory)
-    : m_view(view), m_messageHandler(messageHandler),
-      m_batchPresenterFactory(std::move(batchPresenterFactory)),
-      m_instrument() {
+    : m_view(view), m_messageHandler(messageHandler), m_instrument(),
+      m_slitCalculator(std::move(slitCalculator)),
```

Comment:
Is it safe to use `std::move` with `std::unique_ptr`?

Reply:
Yes, this transfers ownership to the new unique_ptr and resets the old one